File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int n=nums.size();
//         appears more than n/3 times
        map<int,int> mp;
        for(int x:nums){
            mp[x]++;
        }
        vector<int> ans;
        int vals = floor(n/3);
        for(auto x:mp){
            if(x.second > vals){
                ans.push_back(x.first);
            }
        }
        return ans;
    }
};

File: 0229-majority-element-ii/0229-majority-element-ii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var majorityElement = function(nums) {
   let n = nums.length;
   let mp = new Map();
   for(const x of nums){
       if(mp.has(x)){
           mp.set(x,mp.get(x)+1);
       }
       else{
           mp.set(x,1);
       }
   }
    let ans = [];
    const vals = Math.floor(n/3);
    for(const [key,value] of mp.entries()){
        if(value>vals){
            ans.push(key);
        }
    }
    return ans;
};




